ethcontract-derive/src/lib.rs

```diff
@@ -26,69 +27,166 @@ use syn::{
 ///
-/// Currently the proc macro accepts additional parameters to configure some
-/// aspects of the code generation. Specifically it accepts:
-/// - `crate`: The name of the `ethcontract` crate. This is useful if the crate
-///   was renamed in the `Cargo.toml` for whatever reason.
-/// - `contract`: Override the contract name that is used for the generated
-///   type. This is required when using sources that do not provide the contract
-///   name in the artifact JSON such as Etherscan.
-/// - `mod`: The name of the contract module to place generated code in. Note
-///   that the root contract type gets re-exported in the context where the
-///   macro was invoked. This defaults to the contract name converted into snake
-///   case.
-/// - `deployments`: A list of additional addresses of deployed contract for
-///   specified network IDs. This mapping allows `MyContract::deployed` to work
-///   for networks that are not included in the Truffle artifact's `networks`
-///   property. Note that deployments defined this way **take precedence** over
-///   the ones defined in the Truffle artifact. This parameter is intended to be
-///   used to manually specify contract addresses for test environments, be it
-///   testnet addresses that may defer from the originally published artifact or
-///   deterministic contract addresses on local development nodes.
-/// - `methods`: A list of mappings from method signatures to method names
-///   allowing methods names to be explicitely set for contract methods. This
-///   also provides a workaround for generating code for contracts with multiple
-///   methods with the same name.
-/// - `event_derives`: A list of additional derives that should be added to
+/// Currently, the proc macro accepts additional parameters to configure some
+/// aspects of the code generation. Specifically it accepts the following.
+///
+/// - `format`: format of the artifact.
+///
+///   Available values are:
+///
+///   - `truffle` (default) to use [truffle loader];
+///   - `hardhat` to use [hardhat loader] in [single export mode];
+///   - `hardhat_multi` to use hardhat loader in [multi export mode].
+///
+///   Note that hardhat artifacts export multiple contracts. You'll have to use
+///   `contract` parameter to specify which contract to generate bindings to.
+///
+///   [truffle loader]: ethcontract_common::artifact::truffle::TruffleLoader
+///   [hardhat loader]: ethcontract_common::artifact::hardhat::HardHatLoader
+///   [single export mode]: ethcontract_common::artifact::hardhat::Format::SingleExport
+///   [multi export mode]: ethcontract_common::artifact::hardhat::Format::MultiExport
+///
+/// - `contract`: name of the contract we're generating bindings to.
+///
+///   If an artifact exports a single unnamed artifact, this parameter
+///   can be used to set its name. For example:
+///
+///   ```ignore
+///   contract!(
+///       "etherscan:0xC02AAA39B223FE8D0A0E5C4F27EAD9083C756CC2",
```

Comment:
Here too :stuck_out_tongue: 
```suggestion
///       "etherscan:0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
```